src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+
+    private String name = "Square";
+    private double firstLeg, secondLeg;
+    private String color;
+
+    public Square(double firstLeg, String color) {
```

Comment:
The constructor should not have a 'color' parameter as color should be generated using the 'ColorSupplier' class.